diff --git a/services/core/java/com/android/server/ExtconStateObserver.java b/services/core/java/com/android/server/ExtconStateObserver.java
index 92f30ff50112..9e99f3b9415f 100644
--- a/services/core/java/com/android/server/ExtconStateObserver.java
+++ b/services/core/java/com/android/server/ExtconStateObserver.java
@@ -53,6 +53,9 @@ public abstract class ExtconStateObserver<S> extends ExtconUEventObserver {
     public void onUEvent(ExtconInfo extconInfo, UEvent event) {
         if (LOG) Slog.d(TAG, extconInfo.getName() + " UEVENT: " + event);
         String name = event.get("NAME");
+        if(name == null) {
+            return;
+        }
         S state = parseState(extconInfo, event.get("STATE"));
         if (state != null) {
             updateState(extconInfo, name, state);
diff --git a/services/core/java/com/android/server/WiredAccessoryManager.java b/services/core/java/com/android/server/WiredAccessoryManager.java
index 6a808b813bef..b53e45a7e8e3 100644
--- a/services/core/java/com/android/server/WiredAccessoryManager.java
+++ b/services/core/java/com/android/server/WiredAccessoryManager.java
@@ -556,6 +556,9 @@ final class WiredAccessoryManager implements WiredAccessoryCallbacks {
 
             if (name == null) {
                 name = event.get("SWITCH_NAME");
+                if(name == null) {
+                    return;
+                }
             }
 
             try {
